### PR TITLE
Z sampler now do not need the whole z

### PR DESCRIPTION
**Pull request by FrancescaCrucinio**

### CODE REVIEW

**Comment by TomasVaskevicius** [comment]
Let's wait until we have the FireFly algorithm working before making it faster. This can only add more bugs on top what we already have (but don't close this pull request).

File: firefly/R/explicit_resampling.R [hunk 1]
```diff
@@ -7,27 +7,31 @@
...
+    #       switch on/off and if they should be turned off or on
 
     N <- model$number.of.data.points
+    # Build a matrix containing indexes to update and updated values
+    # In the fist column we put indexes to update
+    # In the second column the updated value
+    z.updates <- matrix(0,nrow=floor(N*alpha), ncol=2)
     # Sample a fraction of fireflies to update.
-    state.change.indexes <- sample.int(N, size = floor(N*alpha), replace = FALSE)
+    z.updates[,1] <- sample.int(N, size = length(z.updates[,1]), replace = FALSE)
```
Comment (FrancescaCrucinio) [inline firefly/R/explicit_resampling.R]: size = nrow(z.updates)

File: firefly/R/explicit_resampling.R [hunk 2]
```diff
@@ -7,27 +7,31 @@
...
     # Update the selected fireflies.
-    z[state.change.indexes] <- ifelse(
-      log(runif(length(state.change.indexes))) <=
-        model$LogBound(state.change.indexes, theta)
-        - model$LogLikelihood(state.change.indexes, theta),
+    z.updates[,2] <- ifelse(
+      log(runif(length(z.updates[,1]))) <=
```
Comment (FrancescaCrucinio) [inline firefly/R/explicit_resampling.R]: nrow(z.updates)

File: firefly/R/mh_resampling.R [hunk 3]
```diff
@@ -9,26 +9,30 @@
...
+  # In the second column the updated value
+  z.updates <- matrix(0,nrow=floor(N*alpha), ncol=2)
   # Sample a random fraction of fireflies to update
   n.state.changes <- sample.int(N,1)
-  state.change.indexes <- sample.int(N, size=n.state.changes, replace=FALSE)
+  z.updates[,1] <- sample.int(N, size = n.state.changes, replace = FALSE)
   
   # Prosed on to off switches
-  on.to.off <- which(z[state.change.indexes]>0)
+  on.to.off <- which(z.updates[,1]>0)
```
Comment (FrancescaCrucinio) [inline firefly/R/mh_resampling.R]: non sense

Comment (FrancescaCrucinio) [reply firefly/R/mh_resampling.R]: z.updates are all >0 because are indexes!!!
we need to pass z to this algorithm

**Pull request closed without merging**